schema_based_on_diagram.sql

```diff
@@ -0,0 +1,57 @@
+-- create medical database
+CREATE DATABASE medical
+
+-- create patients table
+CREATE TABLE "patients" (
+  "id" INT,
+  "name" VARCHAR(60),
+  "date_of_birth" DATE,
+  PRIMARY KEY ("id")
+);
+
+-- create medical_histories table
+CREATE TABLE "medical_histories" (
+  "id" INT,
+  "admitted" TIMESTAMP,
+  "patient_id" INT,
+  "status" VARCHAR(60),
+  PRIMARY KEY ("id"),
+  CONSTRAINT "FK_medical_histories.patient_id"
+    FOREIGN KEY ("patient_id")
+      REFERENCES "patients"("id")
+);
+
+-- create treatments table
+CREATE TABLE "treatments" (
+  "id" INT,
+  "type" VARCHAR(60),
+  "name" VARCHAR(60),
+  PRIMARY KEY ("id")
+);
+
+-- create invoices table
+CREATE TABLE "invoices" (
+  "id" INT,
+  "total_amount" DECIMAL,
+  "generated_at" TIMESTAMP,
+  "payed_at" TIMESTAMP,
+  "medical_history_id" INT,
+  PRIMARY KEY ("id")
+);
+
+-- create invoice_items table
+CREATE TABLE "invoice_items" (
+  "id" INT,
+  "unit_price" DECIMAL,
+  "quantity" INT,
+  "total_price" DeCIMAL,
+  "invoice_id" INT,
+  "treatment_id" INT,
+  PRIMARY KEY ("id"),
+  CONSTRAINT "FK_invoice_items.treatment_id"
+    FOREIGN KEY ("treatment_id")
+      REFERENCES "treatments"("id"),
+  CONSTRAINT "FK_invoice_items.invoice_id"
+    FOREIGN KEY ("invoice_id")
+      REFERENCES "invoices"("id")
+);

```

Comment:
- Please a table to handle `many-to-many` relationship between `Treatments` and `Medical_histories` tables. You can do something like:
```ruby
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

Reply:
@afanimartin Thanks for your observation and suggestion. The file has been updated.